models/issue.go

```diff
@@ -1315,6 +1320,14 @@ func applyReviewRequestedCondition(sess *xorm.Session, reviewRequestedID int64)
 			reviewRequestedID, ReviewTypeApprove, ReviewTypeReject, ReviewTypeRequest, reviewRequestedID)
 }
 
+func applySubscribedCondition(sess *xorm.Session, subscribedID int64) *xorm.Session {
+	return sess.And("(issue.id IN (SELECT issue_id FROM issue_watch WHERE (is_watching = ?) AND user_id = ?)) "+
+		"OR ((issue.id IN (SELECT issue_id FROM comment WHERE poster_id = ?)) AND (NOT issue.id IN (SELECT issue_id FROM issue_watch WHERE user_id = ? AND (is_watching = ?)))) "+
+		"OR ((issue.id IN (SELECT id FROM issue WHERE poster_id = ?)) AND (NOT issue.id IN (SELECT issue_id FROM issue_watch WHERE user_id = ? AND (is_watching = ?)))) "+
+		"OR (issue.repo_id IN (SELECT id FROM watch WHERE user_id = ? AND mode = ?) AND (NOT issue.id IN (SELECT issue_id FROM issue_watch WHERE user_id = ? AND (is_watching = ?))))",
+		true, subscribedID, subscribedID, subscribedID, false, subscribedID, subscribedID, false, subscribedID, true, subscribedID, false)
```

Comment:
It's really hard to understand the logic here, and if there is a performance problem (eg: slow query), how should we deal with it?

Reply:
Logic: an issue is "accepted" if one of the following is true:
1. It is in the `issue_watch` table and `is_watching` is true
2. The user commented on or created the issue and it is **not** in the values in the `issue_watch` table where `is_watching` is false
3. The user watches the repo and it is **not** in the values in the `issue_watch` table where `is_watching` is false

I know that it's a complex SQL statement, but there isn't really a way to change it and get all of the subscribed issues. 


Performance: I know this problem and it is hard to deal with it. This is one of the reasons why this implementation is better than the previous one, which runs this every time you open the issues/prs dashboard pages or the repo issues/prs pages. With this, there is an own page for it and it only gets queried if this page is opened. Unfortunately, it is not easy to get every subscription because of Gitea's db structure.

Reply:
Maybe we can wait for some more experienced developers' suggestions ....

Reply:
Just a comment on this: the mechanism to find out if a user watches an issue uses is the same - it's just more simple, but it also checks this [here](https://github.com/go-gitea/gitea/blob/3b06675811cb3cce71447c47434661757e1ea38a/models/issue.go#L1885) and [here](https://github.com/go-gitea/gitea/blob/47193dbcd97555d0c72d3219a372d9d5d01755d3/models/issue_watch.go#L86).

Reply:
My question was asked when I was new to Gitea, and needed more helps from experienced developers.

Now, after reading more Gitea's SQL code, I think it might be  acceptable. No problem from my side.